src/lib.rs

```diff
@@ -221,32 +221,44 @@ fn run_physics_schedule(world: &mut World) {
         .remove_resource::<XpbdLoop>()
         .expect("no xpbd loop resource");
 
-    if xpbd_loop.paused {
-        xpbd_loop.accumulator += DELTA_TIME * xpbd_loop.queued_steps as Scalar;
-        xpbd_loop.queued_steps = 0;
-    } else {
-        let time = world.resource::<Time>();
+    #[cfg(feature = "f32")]
+    let delta_seconds = world.resource::<Time>().delta_seconds();
+    #[cfg(feature = "f64")]
+    let delta_seconds = world.resource::<Time>().delta_seconds_f64();
 
-        #[cfg(feature = "f32")]
-        let delta_time = time.delta_seconds();
+    let time_step = *world.resource::<PhysicsTimestep>();
 
-        #[cfg(feature = "f64")]
-        let delta_time = time.delta_seconds_f64();
+    // Update `DeltaTime` according to the `PhysicsTimestep` configuration
+    let dt = match time_step {
+        PhysicsTimestep::Fixed(fixed_delta_seconds) => fixed_delta_seconds,
+        PhysicsTimestep::Variable { max_dt } => delta_seconds.min(max_dt),
```

Comment:
I'm still having some trouble understanding how this should work, but why would the simulation slow down when the framerate drops? The simulation is progressed by `delta_seconds` every frame regardless of framerate, so I don't think it should slow down (unless you mean performance). The clamping just makes sure that the length of each individual substep doesn't exceed `max_dt` to avoid large jumps in the simulation when the framerate is low.

Currently the code doesn't consume the remainder though, so the simulation could be slightly behind on some frames. I tried consuming the remainder, but it seems to cause the simulation to explode. I wonder if the remainder is so small that it causes floating point precision errors, and we should clamp the lower bound as well?

Reply:
Might be I read through the code to quickly. I'll try to read through it again and maybe word it better

Reply:
Yeah, think I just read it wrong. Likely got confused by `dt` and `delta_seconds` being a bit similar or something.

Strange that it would blow up with fewer substeps...

Reply:
Yeah the names are probably a bit confusing, I might change them or at least add comments if I come up with something clearer.

For consuming the remainder, is this what you meant? It didn't work in my testing

```rs
// Consume the remainder of the accumulator
if xpbd_loop.accumulator > 0.0 {
    // This works
    world.resource_mut::<DeltaTime>().0 = 0.01;

    // This explodes
    world.resource_mut::<DeltaTime>().0 = xpbd_loop.accumulator;

    world.run_schedule(XpbdSchedule);
    xpbd_loop.accumulator = 0.0;
}
```

Reply:
Wouldn't you also need to set num substeps accordingly, or am I missing something again?